### PR TITLE
Switch to keybase go-crypto (for some elliptic curve key) + test

### LINKED ISSUE #1898: GPG2 ECDSA brainpool curve key not working
- Gitea version (or commit ref): 295f560
- Git version: 2.7.4
- Operating system: x86_64 GNU/Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (https://try.gitea.io/user/settings/keys)
  - [ ] No
  - [ ] Not relevant

## Description

When I try to add my GPG2 armored pubkey, gitea says `Sorry, we were not able to verify your GPG key: failed to parse gpg key openpgp: unsupported feature: invalid oid length: 9
`. In my case I used the `brainpoolP256r1` curve.

I'm really sure it's because ECDSA with a brainpool curve.

## Screenshots

[Issue](http://imgur.com/svXH8fK)

### PR DESCRIPTION
**Pull request by sapk**
Fix #1898 by using keybase/go-crypto in place of golang.org/x/crypto.

In folder openpgp, keybase/go-crypto is not that far away of golang.org/x/crypto and implement more key type.

### CODE REVIEW

**Comment by bkcsoft** [comment]
LGTM. Though I'd prefer go-crypto/openpgp to be merged into stdlib 😉

File: models/gpg_key_test.go [hunk 1]
```diff
@@ -47,6 +47,27 @@ MkM/fdpyc2hY7Dl/+qFmN5MG5yGmMpQcX+RNNR222ibNC1D3wg==
...
+muCtPCUg9hlWAP4yq6M78NW8STfsXgn6oeziMYiHSTmV14nOamLuwwDWM7hXBFeh
+8JISCSskAwMCCAEBBwIDBG3A+XfINAZp1CTse2mRNgeUE5DbUtEpO8ALXKA1UQsQ
+DLKq27b7zTgawgXIGUGP6mWsJ5oH7MNAJ/uKTsYmX40DAQgHiGcEGBMIAA8FAleh
+8JICGwwFCQlmAYAACgkQa4lN38MEzOZwKAD/QKyerAgcvzzLaqvtap3XvpYcw9tc
+OyjLLnFQiVmq7kEA/0z0CQe3ZQiQIq5zrs7Nh1XRkFAo8GlU/SGC9XFFi722
+=ZiSe
+-----END PGP PUBLIC KEY BLOCK-----`
+
+	key, err := checkArmoredGPGKeyString(testGPGArmor)
+	assert.Nil(t, err, "Could not parse a valid GPG armored key", key)
```
Comment (ethantkoenig) [inline models/gpg_key_test.go]: For checking errors, you should use `assert.NoError(..)`

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

**Review by ethantkoenig — APPROVED** [review]

File: models/gpg_key_test.go (hunk 1 above)
Comment (sapk) [reply models/gpg_key_test.go]: done and update all test in this file to use .Error .NotError

**Comment by Bwko** [comment]
LGTM

**Pull request merged**